cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,40 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import HomePageObject from '../support/pages/home.pageObject';
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
+  let user;
+
   before(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
+  });
 
+  beforeEach(() => {
+    signUpPage.visit();
   });
 
-  it('should ...', () => {
+  it('should provide an ability to register with valid credentials', () => {
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
 
+    homePage.assertHeaderContainUsername(user.username);
   });
+
+  it.only('should not provide an ability to register with wrong credentials',
+    () => {
+      signUpPage.typeEmail('sdf');
+      signUpPage.typePassword('ssf');
```

Comment:
The test for unsuccessful registration uses invalid email and password formats ('sdf' and 'ssf'). Ensure that the `signUpPage.assertFailedSignUp()` method is correctly implemented to check for the expected error messages or conditions that occur when invalid credentials are used. This will help verify that the application properly handles invalid input.